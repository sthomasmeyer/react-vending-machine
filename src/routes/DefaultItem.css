.DefaultItem {
  margin-left: 10px;
}
