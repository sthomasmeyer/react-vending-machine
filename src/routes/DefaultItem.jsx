import React from 'react';
import './DefaultItem.css';

const DefaultItem = ({ param }) => {
  return (
    <div className='DefaultItem'>
      <p>Select the {param} that you're interested in.</p>
    </div>
  );
};

export default DefaultItem;
