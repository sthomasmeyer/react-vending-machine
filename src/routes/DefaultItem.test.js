import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import DefaultItem from './DefaultItem';

//Smoke test:
it('renders without crashing', () => {
  render(<DefaultItem />);
});

it('matches snapshot', () => {
  const { asFragment } = render(<DefaultItem />);
  expect(asFragment()).toMatchSnapshot();
});

it('generates expected message w/ given parameters', () => {
  render(<DefaultItem param={'snack'} />);
  expect(
    screen.getByText(`Select the snack that you're interested in.`)
  ).toBeInTheDocument();
});
