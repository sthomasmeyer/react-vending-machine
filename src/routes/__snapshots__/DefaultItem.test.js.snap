// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="DefaultItem"
  >
    <p>
      Select the  that you're interested in.
    </p>
  </div>
</DocumentFragment>
`;
